src/ast/ast.h

```diff
@@ -111,9 +111,11 @@ class Binop : public Expression {
 
 class Unop : public Expression {
 public:
-  Unop(int op, Expression *expr) : expr(expr), op(op) { }
+  Unop(int op, Expression *expr, bool is_post_op = false)
```

Comment:
I like this approach of using the unary operator. I had considered something similar but couldn't quite get it figured out